dask_expr/_core.py

```diff
@@ -279,9 +300,12 @@ def simplify_once(self, dependents: defaultdict):
         expr:
             output expression
         """
+        if self.fully_simplified:
+            return self
```

Comment:
So, if this exact expression has every been simplified and it wasn't changed, then we return early here, correct? Even if the last time it was simplified, a different `dependents` argument was used?

I don't have any reason to believe this is "wrong" right now - Just want to make sure I understand. Is it possible for `dependents` to change whether or not a specific expression gets changed?

Reply:
This isn't true if any dependents changed unless you mutate the expression in place which I hope we're not doing anywhere. This is only set if the object already passed through simplify_once without any modifications

Reply:
I'll remove this again. The singleton is the important change and I don't want to get stuck on this caching issue

Reply:
I get why this exists but I'm confused why this has to be provided externally. Doesn't matter right now. This is a problem for another issue